pandas/src/inference.pyx

```diff
@@ -1132,15 +1132,32 @@ def map_infer(ndarray arr, object f, bint convert=1):
     return result
 
 
-def to_object_array(list rows):
+def to_object_array(list rows, int width=0):
+    """
+    Convert a list of lists into an object array.
+
+    Parameters
+    ----------
+    rows : list
+        A list of lists to be converted into an array
+    width : int
+        The minimum width of the object array. If a list
+        in `rows` contains fewer than `width` elements,
+        the remaining elements in the corresponding row
+        will all be `NaN`.
+
+    Returns
+    -------
+    obj_array : numpy array of the object dtype
+    """
     cdef:
         Py_ssize_t i, j, n, k, tmp
         ndarray[object, ndim=2] result
         list row
 
     n = len(rows)
 
-    k = 0
+    k = width
```

Comment:
well what *should* you do here? e.g. you are specifying width, which could be less < len(row).?